sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -274,12 +275,30 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
   }
 
   object CartesianProduct extends Strategy {
+    def getSmallSide(left: LogicalPlan, right: LogicalPlan): BuildSide = {
+      if (right.statistics.sizeInBytes < left.statistics.sizeInBytes) {
+        joins.BuildRight
+      } else {
+        joins.BuildLeft
+      }
+    }
```

Comment:
Agree with @yhuai and @chenghao-intel. Just would like to provide a concrete example: compression ratio of Parquet can be easily ten times of JSON. When joining a Parquet table and a JSON table, you may even end up with a much worse situation.